WcaOnRails/app/models/competition.rb

```diff
@@ -458,11 +465,82 @@ def in_progress?
   end
 
   def is_over?
-    end_date < Date.today
+    !!end_date && end_date < Date.today
+  end
+
+  def results_updated_at
+    Option.from_nilable(results.order('updated_at desc').pluck(:updated_at).first)
+  end
+
+  def result_cache_key(view)
+    [id, view, results_updated_at.map(&:iso8601).unwrap_or("")]
```

Comment:
I just removed `results_updated_at` completely because falling back to `start_date` doesn't make much sense.